htdocs/societe/class/societe.class.php

```diff
@@ -1479,7 +1479,7 @@ public function update($id, $user = '', $call_trigger = 1, $allowmodcodeclient =
 
 			$sql .= ",fk_effectif = ".($this->effectif_id > 0 ? ((int) $this->effectif_id) : "null");
 			if (isset($this->stcomm_id)) {
-				$sql .= ",fk_stcomm=".($this->stcomm_id > 0 ? ((int) $this->stcomm_id) : "0");
+				$sql .= ",fk_stcomm=".($this->stcomm_id >= -1 ? ((int) $this->stcomm_id) : "0");
```

Comment:
This test is strange, it seems it will ba always true.

Reply:
@eldy it is especially that it would not have been necessary to put the value -1 for status "Do not contact", it would have been better to put 9 !

Reply:
@eldy i send a better fix